lib/hanami/action/contract.rb

```diff
@@ -0,0 +1,139 @@
+# frozen_string_literal: true
+
+module Hanami
+  class Action
+    # A wrapper for defining validation rules using Dry::Validation. This class essentially
+    # wraps a Dry::Validation::Contract and acts as a proxy to actually use Dry gem
+    #
+    # Defined via the `contract` block in an action class.
+    # Although more complex domain-specific validations, or validations concerned with things such as uniqueness
+    # are usually better performed at layers deeper than your HTTP actions, Contract still provides helpful features
+    # that you can use without contravening the advice form above.
+    #
+    # @since 2.2.0
+    class Contract
+      # A wrapper for the result of a contract validation
+      # @since 2.2.0
+      # @api private
+      class Result < SimpleDelegator
+        # @since 2.0.0
+        # @api private
+        def output
+          __getobj__.to_h
+        end
```

Comment:
This should be called `to_h` too, not sure why we have `output` here, in dry-schema/validation it's private API